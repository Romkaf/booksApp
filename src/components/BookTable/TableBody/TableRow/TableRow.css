.root {
	display: grid;
	grid-template-columns: repeat(auto-fit, minmax(50px, 1fr));
	border-bottom: 1px solid var(--grey);
}

.root:last-child {
	border-bottom: none;
}

.ceil {
	padding: 0.75rem;
	word-break: break-word;
}

@media (max-width: 500px) {
	.ceil {
		padding: 0.35rem 0.25rem;
		font-size: 16px;
	}
}
