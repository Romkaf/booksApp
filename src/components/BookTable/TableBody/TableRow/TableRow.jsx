import React from 'react';
import styles from './TableRow.css';
import PropTypes from 'prop-types';

const TableRow = ({ item }) => {
	const { root, ceil } = styles;
	const { title, authors, date, publisher } = item;
	const dataArray = [title, authors, date, publisher];

	return (
		<div className={root}>
			{dataArray.map((it, idx) => (
				<div className={ceil} key={idx}>
					{it}
				</div>
			))}
		</div>
	);
};

TableRow.propTypes = {
	item: PropTypes.shape({
		title: PropTypes.string,
		authors: PropTypes.string,
		publisher: PropTypes.string,
		date: PropTypes.string,
	}),
};

export default TableRow;
