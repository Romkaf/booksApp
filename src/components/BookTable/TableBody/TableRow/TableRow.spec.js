import React from 'react';
import TableRow from './TableRow';
import { shallow } from 'enzyme';

describe('TableRow', () => {
	const item = {
		id: 1,
		title: 'test',
		authors: 'test',
		publisher: 'publisher',
		date: 'date',
	};

	it('should render component', () => {
		const component = shallow(<TableRow item={item} />);
		expect(component).toMatchSnapshot();
	});
});
