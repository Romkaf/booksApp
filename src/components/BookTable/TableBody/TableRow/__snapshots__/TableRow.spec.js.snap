// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow should render component 1`] = `
<tr>
  <td
    className="ceil"
  >
    test
  </td>
  <td
    className="ceil"
  >
    test
  </td>
  <td
    className="ceil"
  >
    publisher
  </td>
  <td
    className="ceil"
  >
    date
  </td>
</tr>
`;
